### ISSUE TITLE
Manually configuring choice of sd-bus implementation library

### ISSUE THREAD

**Issue by lch361** [Kistler-Group/sdbus-cpp#399]
While I appreciate that this library can work without systemd or elogind, there is still a problem. For example, when you have elogind and basu installed on your system simultaneously, there is no way for user to specify using basu - cmake will always choose elogind for him.

If I were an expert in cmake, I would add an option:
```sh
cmake -DIMPL=basu     # builds sdbus-c++ with basu, ignoring presence of systemd & elogind
cmake -DIMPL=systemd  # builds sdbus-c++ with systemd, ignoring presence of elogind & basu
cmake -DIMPL=elogind  # builds sdbus-c++ with elogind, ignoring presence of systemd & basu
cmake -DIMPL=something_else  # configuration error
cmake  # builds with what is present on a system
```
(or maybe another name instead of `IMPL`)

**Comment by sangelovic** [comment]
@lch361 Thanks for the feedback. Feel free to review #409, which implements your proposal.

**Comment by sangelovic** [comment]
The mentioned PR is for currently prepared `v2.0.0` release, which will soon go to master. Is that OK for you? Or shall we backport also to `v1.x` (currently master)?

**Comment by lch361** [comment]
> The mentioned PR is for currently prepared `v2.0.0` release, which will soon go to master. Is that OK for you? Or shall we backport also to `v1.x` (currently master)?

Do as you want on this one

**Comment by sangelovic** [comment]
Fixed in #409, which being part of #364 will soon be released as `v2.0.0`
